src/test/features/claim/routes/claimant-company-details.ts

```diff
@@ -0,0 +1,114 @@
+import { expect } from 'chai'
+import * as request from 'supertest'
+import * as config from 'config'
+
+import { attachDefaultHooks } from '../../../routes/hooks'
+import '../../../routes/expectations'
+import { checkAuthorizationGuards } from './checks/authorization-check'
+import { CompanyDetails } from 'app/forms/models/companyDetails'
+import { Paths as ClaimPaths } from 'claim/paths'
+import { Address } from 'forms/models/address'
+import { app } from '../../../../main/app'
+
+import * as idamServiceMock from '../../../http-mocks/idam'
+import * as draftStoreServiceMock from '../../../http-mocks/draft-store'
+
+const cookieName: string = config.get<string>('session.cookieName')
+const input = {
+  name: 'Anirudha Inc.',
+  type: 'company',
+  contactPerson: 'John Snow',
+  address: { line1: 'Apartment 99', line2: '', city: 'London', postcode: 'SE28 0JE' } as Address,
+  hasCorrespondenceAddress: false
+} as CompanyDetails
+
+describe('claimant as company details page', () => {
+  attachDefaultHooks()
+
+  describe('on GET', () => {
+    checkAuthorizationGuards(app, 'get', ClaimPaths.claimantCompanyDetailsPage.uri)
+
+    it('should render page when everything is fine', async () => {
+      idamServiceMock.resolveRetrieveUserFor(1, 'cmc-private-beta', 'claimant')
+      draftStoreServiceMock.resolveRetrieve('claim')
+
+      await request(app)
+        .get(ClaimPaths.claimantCompanyDetailsPage.uri)
+        .set('Cookie', `${cookieName}=ABC`)
+        .expect(res => expect(res).to.be.successful.withText('Company details'))
+    })
+  })
+
+  describe('on POST', () => {
+    checkAuthorizationGuards(app, 'post', ClaimPaths.claimantCompanyDetailsPage.uri)
+
+    describe('for authorized user', () => {
+      beforeEach(() => {
+        idamServiceMock.resolveRetrieveUserFor(1, 'cmc-private-beta', 'claimant', 'defendant')
+      })
+
+      it('should render page with error when claimant name is invalid', async () => {
```

Comment:
These tests are not supposed to test validation rules but ensure overall behaviour upon validation errors is correct. To test validation rules we use to write validation unit tests close to validation rules.